app/scripts/controllers/metametrics.ts

```diff
@@ -409,11 +418,30 @@ export default class MetaMetricsController {
     const { fragments } = this.store.getState();
 
     const id = options.uniqueIdentifier ?? uuidv4();
-    const fragment = {
+    let fragment = {
       id,
       ...options,
       lastUpdated: Date.now(),
     };
+
+    /**
+     * HACK: "transaction-submitted-<id>" fragment hack
+     * A "transaction-submitted-<id>" fragment may exist following the "Transaction Added"
+     * event to persist accumulated event fragment props to the "Transaction Submitted" event
+     * which fires after a user confirms a transaction. Rejecting a confirmation does not fire the
+     * "Transaction Submitted" event. In this case, these abandoned fragments will be deleted
+     * instead of finalized with canThrowAwayIfAbandoned set to true.
+     */
+    const hasExistingSubmittedFragment =
+      options.initialEvent === TransactionMetaMetricsEvent.submitted &&
```

Comment:
oops, sorry I mixed up the [comment below](https://github.com/MetaMask/metamask-extension/pull/28240#discussion_r1827500220) with this I think it could be a good solution. 

I think due to timing and priorities, we can keep this solution. I attempted this solution for a bit and got stuck on the garbage collection logic. I'll add a proposal C above for decoupling these props from the submitted event